sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -82,12 +84,16 @@ struct _pi_device : _pi_object {
 struct _pi_context : _pi_object {
   _pi_context(pi_device ArgDevice) : Device{ArgDevice} {}
 
-  /// One-to-one mapping between Context and Device
+  // One-to-one mapping between Context and Device
   pi_device Device;
 
-  /// Map SVM memory starting address to corresponding
-  /// CmBufferSVM object. CmBufferSVM object is needed to release memory.
+  // Map SVM memory starting address to corresponding
+  // CmBufferSVM object. CmBufferSVM object is needed to release memory.
   std::unordered_map<void *, cm_support::CmBufferSVM *> Addr2CmBufferSVM;
+  // A lock guarding access to Addr2CmBufferSVM
+  std::mutex Addr2CmBufferSVMLock;
```

Comment:
Interesting if synchronized map access could be encapsulated here, not scattered around plugin (?)

Reply:
Will discuss with similar access controls (https://github.com/intel/llvm/issues/5599)

Reply:
Let's make it a separate PR in any case.

Reply:
I'll post another PR for discussing this.